osu.Game/Skinning/SkinImporter.cs

```diff
@@ -43,6 +46,34 @@ public SkinImporter(Storage storage, RealmAccess realm, IStorageResourceProvider
 
         private const string unknown_creator_string = @"Unknown";
 
+        /// <summary>
+        /// Update an existing skin with the contents of a path
+        /// </summary>
+        /// <param name="notification">The progress notification</param>
+        /// <param name="task">The <see cref="ImportTask"/> to update the <paramref name="original"/> with</param>
+        /// <param name="original">The <see cref="SkinInfo"/> to update</param>
+        /// <returns></returns>
+        public override Task<Live<SkinInfo>?> ImportAsUpdate(ProgressNotification notification, ImportTask task, SkinInfo original)
+        {
+            var skinInfoLive = original.ToLive(Realm);
+
+            skinInfoLive.PerformWrite(skinInfo =>
+            {
+                skinInfo.Files.Clear();
+
+                string[] filesInMountedDirectory = Directory.EnumerateFiles(task.Path, "*.*", SearchOption.AllDirectories).Select(f => Path.GetRelativePath(task.Path, f)).ToArray();
+
+                foreach (string file in filesInMountedDirectory)
+                {
+                    using var stream = File.OpenRead(Path.Combine(task.Path, file));
+
+                    modelManager.AddFile(original, stream, file);
+                }
+            });
```

Comment:
The process should probably match the beatmap reimport one, optimally.

Maybe it can shortcut in the case that ini/json files didn't change, if the full import takes too long (although I'd like to hope it doesn't).

Reply:
I was hoping that it would update automatically, since changing the skin.ini and creating a new skin instance for the `SkinInfo` sounds like it should update the skin's name in realm, but it probably isn't because we're reusing the old `SkinInfo` which would have the old `Name` property.

I'll probably end up implementing it as peppy suggested. Before processing the other files, I'll hash the two skin.ini's. If there's a change, I'll investigate what changed and reflect them in realm. If nothing was changed, we're good and we can skip that entire process.

Reply:
I've finally gotten to fixing this, but it's now fixed. See [`165afe3` (#30226)](https://github.com/ppy/osu/pull/30226/commits/165afe357f5a52050f2337e54478f87739f125fb) for the details.